src/game/common/system/file.h

```diff
@@ -26,6 +26,18 @@ struct FileInfo
     int write_time_low;
 };
 
+namespace Thyme
```

Comment:
Do we really need this level of control over the buffering? Can't we just flag filesystem to provide us with a buffered file and accept the standard buffer and leave the File API untouched?

Reply:
Yes. Filesystem user can also just set the BUFFERED flag and then it will use the standard buffer size (512 bytes). The use of `Encode_Buffered_File_Mode` is optional (I use it in `GameTextFile` class for 64 kb size).